src/StructuredLogger/BinaryLogger/BuildEventArgsReader.cs

```diff
@@ -25,7 +25,7 @@ namespace Microsoft.Build.Logging.StructuredLogger
     /// <summary>
     /// Deserializes and returns BuildEventArgs-derived objects from a BinaryReader
     /// </summary>
-    internal partial class BuildEventArgsReader : IBuildEventArgsReaderNotifications, IDisposable
+    public partial class BuildEventArgsReader : IBuildEventArgsReaderNotifications, IDisposable

```

Comment:
Same, can't make this public. But you can add a new public API that wraps this call.